lxd/storage/backend_lxd.go

```diff
@@ -3357,6 +3397,77 @@ func (b *lxdBackend) EnsureImage(fingerprint string, op *operations.Operation) e
 	return nil
 }
 
+// shouldUseOptimizedImage determines if an optimized image should be used based on the provided volume config.
+// It returns true if the volume config aligns with the pool's default configuration, and an optimized image does
+// not exist or also matches the pool's default confgiuration.
+func (b *lxdBackend) shouldUseOptimizedImage(fingerprint string, volConfig map[string]string, op *operations.Operation) (bool, error) {
+	err := b.isStatusReady()
+	if err != nil {
+		return false, err
+	}
+
+	// Skip if the driver doesn't support optimized image volumes.
+	if !b.driver.Info().OptimizedImages {
+		return false, nil
+	}
+
+	// Load image info from database.
+	_, image, err := b.state.DB.Cluster.GetImageFromAnyProject(fingerprint)
+	if err != nil {
+		return false, err
+	}
+
+	// Derive content type from image type. Image types are not the same as instance types, so don't use
+	// instance type constants for comparison.
+	contentType := drivers.ContentTypeFS
```

Comment:
Cant you infer this from the instance type being passed into to `CreateInstanceFromImage`? This would avoid another expensive DB query.